wpiutil/src/main/java/edu/wpi/first/util/RawFrame.java

```diff
@@ -88,6 +88,25 @@ public void setData(ByteBuffer data, int width, int height, int stride, PixelFor
         m_nativeObj, data, data.limit(), width, height, stride, pixelFormat.getValue());
   }
 
+  /**
+   * Set frame data.
+   *
+   * @param data A Java byte[] pointing to the frame data.
+   * @param width The width of the frame, in pixels
+   * @param height The height of the frame, in pixels
+   * @param stride The number of bytes in each row of image data
+   * @param pixelFormat The PixelFormat of the frame
+   */
+  public void setData(byte[] data, int width, int height, int stride, PixelFormat pixelFormat) {
+    m_data = ByteBuffer.wrap(data);
```

Comment:
Does this still incur a copy?

Reply:
No, the ByteBuffer uses the byte[] as backing storage: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/ByteBuffer.html#wrap(byte%5B%5D)